packages/angular_devkit/build_angular/src/utils/i18n-options.ts

```diff
@@ -82,7 +91,79 @@ export function createI18nOptions(
   return i18n;
 }
 
-export function mergeDeprecatedI18nOptions(i18n: I18nOptions, i18nLocale: string | undefined, i18nFile: string | undefined): I18nOptions {
+export async function configureI18nBuild<T extends BrowserBuilderSchema | ServerBuilderSchema>(
+  context: BuilderContext,
+  host: virtualFs.Host<fs.Stats>,
+  options: T,
+): Promise<{
+  buildOptions: T,
+  i18n: I18nOptions,
+}> {
+  if (!context.target) {
+    throw new Error('The builder requires a target.');
+  }
+
+  const buildOptions = { ... options };
+
+  const tsConfig = readTsconfig(buildOptions.tsConfig, context.workspaceRoot);
+  const usingIvy = tsConfig.options.enableIvy !== false;
+  const metadata = await context.getProjectMetadata(context.target);
+  const i18n = createI18nOptions(metadata, buildOptions.localize);
+
+  // Until 11.0, support deprecated i18n options when not using new localize option
+  // i18nFormat is automatically calculated
+  if (buildOptions.localize === undefined && usingIvy) {
+    mergeDeprecatedI18nOptions(i18n, buildOptions.i18nLocale, buildOptions.i18nFile);
+  } else if (buildOptions.localize !== undefined && !usingIvy) {
+    buildOptions.localize = undefined;
+
+    context.logger.warn(`Option 'localize' is not supported with View Engine.`);
+  }
+
+  if (i18n.inlineLocales.size > 0) {
+    // Load locales
+    const loader = await createTranslationLoader();
+
+    const usedFormats = new Set<string>();
+    for (const [locale, desc] of Object.entries(i18n.locales)) {
+      if (i18n.inlineLocales.has(locale)) {
+        const result = loader(desc.file);
+
+        usedFormats.add(result.format);
+        if (usedFormats.size > 1) {
+          // This limitation is technically only for legacy message id support
+          throw new Error(
+            'Localization currently only supports using one type of translation file format for the entire application.',
+          );
+        }
+
+        desc.format = result.format;
+        desc.translation = result.translation;
+      }
+    }
+
+    // Legacy message id's require the format of the translations
+    if (usedFormats.size > 0) {
+      buildOptions.i18nFormat = [...usedFormats][0];
+    }
+  }
+
+  // If inlining store the output in a temporary location to facilitate post-processing
+  if (i18n.shouldInline) {
+    buildOptions.outputPath = fs.mkdtempSync(path.join(fs.realpathSync(os.tmpdir()), 'angular-cli-i18n-'));
+
+    // Remove temporary directory used for i18n processing
+    process.on('exit', async () => {
+      try {
+        rimraf.sync(buildOptions.outputPath);
```

Comment:
Use a separate variable here so that it doesn't close over the whole buildOptions object

Reply:
Good idea